### ISSUE TITLE
named by 'object' inside 'dictionary

### ISSUE THREAD

**Issue by andrea-dintino** [dyne/Zenroom#301]
We need to use an object in a dictionary, whose name we pass as a parameter:

Zencode

```
Scenario 'ecdh': 
Given my name is in a 'string' named 'Sender'
Given that I have my 'keypair'
Given I have a 'string' named 'Recipient' 
Given I have a 'string' named 'Message'

# below the statement needed
Given that I have a 'public key' named by 'Recipient' inside 'Friends'   


When I encrypt the secret message of 'message' for '#Recipient#' 
When I rename the 'secret message' to 'SecretMessage' 

Then print the 'SecretMessage'    
Then print the 'Recipient' 
```

Data

```
{
	"Sender": "Alice",
	"Alice": {
		"keypair": {
			"private_key": "2mZDRS1rE4jT5EuozwZfbS+GLE7ogBfgWOr30wXoe3g=",
			"public_key": "BI/Jwt7PCxfSnypWNH0koLPWOJn0sLMA7VbYlQe6xmIMQms4xvMjLoPXxl4l8d0EzM9ezGVehNCvIHSy+vpctIk="
		}
	},
	"Friends": {
		"Bob": "BJX5HFLhTxd+QeCcywWP4i7QXufoI83j/VvzoaTlfHjJBJeEIhCIUQHIm+paH/aJWHSnAQC0Mea0IiYb7z4Z4bk=",
		"Jenna": "BNRzlJ4csYlWgycGGiK/wgoEw3OizCdx9MWg06rxUBTP5rP9qPASOW5KY8YgmNjW5k7lLpboboHrsApWsvgkMN4="
	},
	"Recipient": "Bob",
	"Message": "Hi, Bob!"
}
```

**Comment by jaromil** [comment]
```
When I encrypt the secret message of 'message' for '#Recipient#' 
```
wrong zencode: `Message` has uppercase and `Recipient` without hashes

**Comment by jaromil** [comment]
```
When I encrypt the secret message of 'Message' for 'Recipient'
```
wrong zencode: `Recipient` should be `Bob` since that is the content passed in the value `Recipient`

perhaps new statement should be:
```
When I encrypt the secret message of 'Message' for recipient named by 'Recipient'
```

**Comment by jaromil** [comment]
Final fix in commit 744a7107
```
Scenario 'ecdh':
Given my name is in a 'string' named 'Sender'
Given that I have my 'keypair'
Given I have a 'string' named 'Recipient'
Given I have a 'string' named 'Message'

# below the statement needed
Given that I have a 'public key' named by 'Recipient' inside 'Friends'

When I rename named by 'Recipient' to 'SecretRecipient'
When I encrypt the secret message of 'Message' for 'SecretRecipient'
When I rename the 'secret message' to 'SecretMessage'

Then print the 'SecretMessage'
Then print the 'SecretRecipient'
```
